src/artemis/experiment_plans/tests/test_full_grid_scan_plan.py

```diff
@@ -146,3 +140,38 @@ def test_detect_grid_and_do_gridscan(
 
         # Check we called out to underlying fast grid scan plan
         mock_fast_grid_scan_plan.assert_called_once_with(ANY, mock_subscriptions)
+
+
+@patch("artemis.experiment_plans.full_grid_scan.grid_detection_plan")
+@patch("artemis.experiment_plans.full_grid_scan.OavSnapshotCallback")
+def test_grid_detection_running_when_exception_raised_then_eiger_unstaged(
+    mock_oav_callback: MagicMock,
+    mock_grid_detection_plan: MagicMock,
+    RE: RunEngine,
+    test_full_grid_scan_params: GridScanWithEdgeDetectInternalParameters,
+    mock_subscriptions: MagicMock,
+    test_config_files: Dict,
+):
+    mock_grid_detection_plan.side_effect = Exception()
+    eiger: EigerDetector = MagicMock(spec=EigerDetector)
+
+    with patch(
+        "artemis.external_interaction.callbacks.fgs.fgs_callback_collection.FGSCallbackCollection.from_params",
+        return_value=mock_subscriptions,
+    ):
+        with pytest.raises(Exception):
+            RE(
+                start_arming_then_do_grid(
+                    parameters=test_full_grid_scan_params,
+                    backlight=MagicMock(),
+                    eiger=eiger,
+                    aperture_scatterguard=MagicMock(),
+                    detector_motion=MagicMock(),
+                    oav_params=OAVParameters("xrayCentring", **test_config_files),
+                )
+            )
+
+        # Check detector was armed
+        eiger.do_arm.set.assert_called_once_with(1)
```

Comment:
It looks like the exception was raised before the detector was armed as this fails with an `AssertionError` because`set` is never called.

```
 FAILED src/artemis/experiment_plans/tests/test_full_grid_scan_plan.py::test_grid_detection_running_when_exception_raised_then_eiger_unstaged - AssertionError: Expected 'set' to be called once. Called 0 times.
```